src/couch/src/couch_file.erl

```diff
@@ -264,6 +265,19 @@ rename_file(Original) ->
         Else -> Else
     end.
 
+rename_dir(RootDelDir, Original, DbName) ->
+    DbDir = binary_to_list(DbName) ++ "_design",
+    [DbPureName | _R] = lists:reverse(filename:split(binary_to_list(DbName))),
```

Comment:
What is this logic doing? Its not making sense to me why we'd be looking at just the last element of the DbName.